kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/BaseOperation.java

```diff
@@ -222,7 +223,9 @@ public T getMandatory() {
     if (item != null) {
       return item;
     }
-    throw new KubernetesClientException("Resource not found!", 404, new StatusBuilder().withCode(404).withMessage("Not found.").build());
```

Comment:
So perhaps we should include details of the source operation in the exception? Wonder if that would be easy enough to do? So we can use the raw message back from kubernetes, but with details of what the caller was trying to do?